jscomp/ml/typecore.ml

```diff
@@ -591,6 +591,62 @@ let build_or_pat env loc lid =
           pat pats in
       (path, rp { r with pat_loc = loc },ty)
 
+let build_or_pat_for_variant_spread env loc lid expected_ty =
+  let path, decl = Typetexp.find_type env lid.loc lid.txt in
+  match decl with
+  | {type_kind = Type_variant constructors} -> (
+    (* TODO: Probably problematic that we don't account for type params here? *)
```

Comment:
@cristianoc did you see this here? That we're not caring about type params. Type params aren't allowed in spreads anyway right now so maybe it doesn't matter.

Reply:
Maybe worth checking that there are no type params, so it's not overlooked in future.